soql-analyzer/src/main/scala/com/socrata/soql/analyzer2/rewrite/RemoveSyntheticColumns.scala

```diff
@@ -0,0 +1,90 @@
+package com.socrata.soql.analyzer2.rewrite
+
+import com.socrata.soql.analyzer2
+import com.socrata.soql.analyzer2._
+import com.socrata.soql.collection._
+
+class RemoveSyntheticColumns[MT <: MetaTypes] private (labelProvider: LabelProvider) extends StatementUniverse[MT] {
+  def rewriteStatement(stmt: Statement): Statement =
+    stmt match {
+      case ct@CombinedTables(_op, _left, _right) =>
+        if(ct.schema.values.exists(_.isSynthetic)) {
+          // there shouldn't ever actually be any synthetic columns to
+          // remove here, so this is "just in case".  We'll add a new
+          // wrapper select that drops the columns, rather than trying
+          // to rewrite left and right in an equivalent way.  Note we
+          // don't care about preserving our own labels, as this is
+          // the output schema for the analysis.
+
+          val newFrom = FromStatement(ct, labelProvider.tableLabel(), None, None)
+          Select(
+            Distinctiveness.Indistinct(),
+            OrderedMap() ++ ct.schema.iterator.flatMap { case (k, v) =>
+              if(v.isSynthetic) {
+                None
+              } else {
+                Some(labelProvider.columnLabel() -> NamedExpr(VirtualColumn[MT](newFrom.label, k, v.typ)(AtomicPositionInfo.Synthetic), v.name, v.hint, isSynthetic = false))
```

Comment:
is this empty columnLabel() what you mean when you say `we don't care about preserving our own labels`?

Reply:
`columnLabel()` is a 0-arg function that returns a fresh label.  But yes - what I mean by that is "before we rewrite, our schema is, say, `label1 -> synthetic text, label2 -> num` and after we rewrite our schema might be `label45 -> num`"